airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalAttemptExecution.java

```diff
@@ -79,7 +98,50 @@ public T get() throws TemporalJobException {
       LOGGER.info("Executing worker wrapper. Airbyte version: {}", EnvConfigs.AIRBYTE_VERSION);
       jobRootDirCreator.accept(jobRoot);
 
-      return execution.apply(jobRoot);
+      final CompletableFuture<T> outputFuture = new CompletableFuture<>();
+
+      final Worker<INPUT, T> worker = workerSupplier.apply(jobRoot);
+
+      final Thread workerThread = new Thread(() -> {
+        mdcSetter.accept(jobRoot, jobId);
+
+        try {
+          final T output = worker.run(inputSupplier.get(), jobRoot);
+          outputFuture.complete(output);
+        } catch (Throwable e) {
+          LOGGER.info("Completing future exceptionally...", e);
+          outputFuture.completeExceptionally(e);
+        }
+      });
+
+      workerThread.start();
+
+      ScheduledExecutorService scheduledExecutor = Executors.newSingleThreadScheduledExecutor();
+      scheduledExecutor.scheduleAtFixedRate(() -> {
+        try {
+          mdcSetter.accept(jobRoot, jobId);
+
+          cancellationHandler.heartbeat(() -> {
```

Comment:
Some questions: 
1. Why does the cancellation handler have a heartbeat method? I would expect it to have a `cancel` method. I don't follow the purpose of the heartbeat here. Is the handler a long-lived process, or is it a function that gets called on demand? 
2. Why does the heartbeat method seem to be running the actual cancellation code?